statsmodels/multivariate/manova.py

```diff
@@ -0,0 +1,240 @@
+# -*- coding: utf-8 -*-
+
+"""Multivariate analysis of variance
+
+author: Yichuan Liu
+"""
+from __future__ import print_function, division
+
+from statsmodels.base.model import Model
+import numpy as np
+from numpy.linalg import matrix_rank, qr
+from statsmodels.iolib import summary2
+from .multivariate_ols import _multivariate_htest
+
+
+def manova_fit(x, y):
+    """
+    For a MANOVA problem y = x * params
+    where y is dependent variables, x is independent variables
+    Perform QR decomposition of x to calculate the sums of squares
+    and cross-products (SSCP) of residuals
+    These are necessary for performing MANOVA hypothesis tests
+
+    No acture
+
+    Parameters
+    ----------
+    x : array-like, each column is a independent variable
+    y : array-like, each column is a dependent variable
+
+    Returns
+    -------
+    a tuple of matrices or values necessary for hypothesis testing
+
+    .. [1] ftp://public.dhe.ibm.com/software/analytics/spss/documentation/statistics/20.0/en/client/Manuals/IBM_SPSS_Statistics_Algorithms.pdf
+
+    """
+    nobs, k_endog = y.shape
+    nobs1, k_exog= x.shape
+    if nobs != nobs1:
+        raise ValueError('x(n=%d) and y(n=%d) should have the same number of '
+                         'rows!' % (nobs1, nobs))
+
+    # Calculate the matrices necessary for hypothesis testing
+    df_resid = nobs - k_exog
+
+    q, r = qr(x)
+    u = q.T.dot(y)
+    sscpr = np.subtract(y.T.dot(y), u.T.dot(u))
+    fittedvalues = u
+    return (df_resid, fittedvalues, sscpr)
+
+
+def manova_htest(hypothesis, fittedvalues, exog_names,
+                            endog_names):
+    """
```

Comment:
also add leading underscore to function name to make it private, it is not directly useful to users